src/Zenova/Platform/Windows.cpp

```diff
@@ -87,6 +89,97 @@ namespace Zenova::PlatformImpl {
 			FreeLibraryAndExitThread(reinterpret_cast<HMODULE>(CleanupVariables), 0);
 		}
 	}
+
+	void* LoadModModuleAndResolveImports(const std::string& module)
+	{
+		void* hModule = LoadLibraryExA(module.c_str(), NULL, DONT_RESOLVE_DLL_REFERENCES);
```

Comment:
I know ``DONT_RESOLVE_DLL_REFERENCES`` [isn't recommend and it might be best if we can avoid it](https://devblogs.microsoft.com/oldnewthing/20050214-00/?p=36463) but have we tried adding all of the mod folders to the search tree with AddDllDirectory before loading a mod? I wonder if this could just be simplified with that feature.

Reply:
DONT_RESOLVE_DLL_REFERENCES is just a way so it doesnt resolve missing references and fail/crash the api, thats why we are resolving the references manually, AddDllDirectory would just make the loader load the dll and not populate the ModInfo and the things from the api, DONT_RESOLVE_DLL_REFERENCES is just a problem if we don't resolve the dependencies manually, and it will be a way to do a future feature that im planning

Reply:
i kinda got some problems trying to load a .net native dll on the mod loader, i kind of know why, thats a problem with this, im not resolving the references in all paths so it gets sus sometimes

Reply:
Having the mod load early isn't really a problem as LoadModule will just get a HANDLE to the already loaded dll

Reply:
> i kinda got some problems trying to load a .net native dll on the mod loader, i kind of know why, thats a problem with this, im not resolving the references in all paths so it gets sus sometimes

this is kinda why I think we should avoid this method

Reply:
yeah, im just thinking about when we go about the minecraft imports system

Reply:
so, what should i do, change it to AddDllDirectory or leave it as it is? it shouldn't break with normal mods at all

Reply:
Doesn't seem like we can utilize this feature with UWP